load_testing/custom_load_testing/testing.py

```diff
@@ -0,0 +1,148 @@
+"""
+Handles the parsing of the config file and running of load tests via Locust commands.
+"""
+
+import logging
+import os
+import shlex
+import subprocess
+import time
+
+logging.basicConfig(level=logging.INFO)
+
+
+def run_single_test(
+    host: str,
+    locustfile: str,
+    users: int,
+    spawn_rate: int,
+    run_time: int,
+    output_subfolder: str,
+    test_html_filpath: str,
+) -> None:
+    """Runs a single locust test.
+
+    This function takes locust parameters and runs a locust load test,
+    saving the results to the output_folder.
+
+    Note:
+    We are using using --autostart and --autoquit instead of --headless since
+    we want to track our tests on the webUI and the webUI must be enabled for
+    plots to be generated for the Locust HTML reports.
+
+    Parameters
+    ----------
+    host : str
+        host to test
+    locustfile : str
+        locustfile to use
+    users : int
+        number of users to simulate
+    spawn_rate : int
+        number of users to spawn per second
+    run_time : str
+        time to run the test for
+    output_subfolder : str
+        Path to output folder
+    test_name : str
+        The name of test
+
+    Returns
+    -------
+    None
+
+    """
+    output_files_root = output_subfolder + "/test"
+
+    locust_command = shlex.split(
+        f"""
+        locust --autostart --autoquit 10
+         --host {host} --locustfile {locustfile}
+         -u {users} -r {spawn_rate} -t {run_time}
+         --csv {output_files_root} --html {test_html_filpath}
+        """
+    )
+    subprocess.run(locust_command)
+
+
+def run_tests(experiment_configs: dict, output_folder: str) -> None:
+    """Runs a load test for each test configuration.
+
+    Takes a dict of experiment parameters and cycles through all values for
+    each parameters, running a load-test for each combination.
+
+    Creates output folders if they don't exist.
+
+    Parameters
+    ----------
+    experiment_configs : dict
+        dict of experiment parameters from the config file
+    output_folder : str
+        Path to output folder
+
+    Returns
+    -------
+    None
+
+    """
+    host = os.getenv("TARGET_AAQ_URL", "localhost")
+    locustfile_list: list = experiment_configs.get("locustfile_list", [])
```

Comment:
update this to read from file ideally

Reply:
I have left is as still reading from the environment but - as mentioned above - variables are now `exported` rather than set in a `.env`. Makes sense especially as there are only two needed (max. of 4 if TypeBot being tested)